File: exercise/java/day09/src/main/java/gift/Child.java
package gift;

public class Child {
    private final String name;
    private final Behavior behavior;
    private final GiftRequest giftRequest;

    public Child(String firstName, String lastName, int age, Behavior behavior, GiftRequest giftRequest) {
        this.name = firstName + " " + lastName;
        this.behavior = behavior;
        this.giftRequest = giftRequest;
    }

    public String getName() {
        return name;
    }

    public Behavior getBehavior() {
        return behavior;
    }

    public GiftRequest getGiftRequest() {
        return giftRequest;
    }
}

File: solution/day09/src/child.ts
import {GiftRequest} from "./giftRequest";

export class Child {
    public readonly name: string;
    public readonly behavior: Behavior;
    public readonly giftRequest: GiftRequest;

    constructor(firstName: string,
                lastName: string,
                age: number,
                behavior: Behavior,
                giftRequest: GiftRequest) {
        this.name = firstName;
        this.behavior = behavior;
        this.giftRequest = giftRequest;
    }
}